corehq/util/quickcache.py

```diff
@@ -87,14 +95,20 @@ def _hash(value, length=32):
         return hashlib.md5(value).hexdigest()[-length:]
 
     def _serialize_for_key(self, value):
-        if isinstance(value, unicode):
-            return 'u' + self._hash(value.encode('utf-8'))
-        elif isinstance(value, str):
-            if len(value) <= 32 and value.isalnum():
-                return 's' + value
+        if isinstance(value, basestring):
+            # Unicode and string values should generate the same key since users generally
+            # intend them to mean the same thing. If a use case for differentiating
+            # them presents itself add a 'lenient_strings=False' option to allow
+            # the user to explicitly request the different behaviour.
+            if isinstance(value, unicode):
+                encoded = value.encode('utf-8')
             else:
-                # for long or non-alphanumeric text
-                return 't' + self._hash(value)
+                try:
+                    encoded = value.decode('utf-8').encode('utf-8')
+                except UnicodeDecodeError:
+                    self.encoding_assert(False, 'Non-utf8 encoded string used as cache vary on')
```

Comment:
I think what I want is to see if this does happen and whether it's likely
to result in problems down the road.